client/thin-replica-client/src/thin_replica_client.cpp

```diff
@@ -96,23 +96,26 @@ void ThinReplicaClient::recordCollectedHash(size_t update_source,
   }
 }
 
-void ThinReplicaClient::readUpdateHashFromStream(size_t server_index,
+bool ThinReplicaClient::readUpdateHashFromStream(size_t server_index,
                                                  HashRecordMap& server_indexes_by_reported_update,
                                                  size_t& maximal_agreeing_subset_size,
                                                  HashRecord& maximally_agreed_on_update) {
   Hash hash;
+  bool has_hash = true;
```

Comment:
We only ever change this variable if we are about to return. So I suggest to simply `return true;` or `return false;` instead of this variable. Given that the return value is documented the code becomes clear.

Reply:
Updated.